widget/widget.go

```diff
@@ -10,33 +10,42 @@ import (
 
 // A base widget class to define the standard widget behaviours.
 type baseWidget struct {
+	sync.RWMutex
 	size     fyne.Size
 	position fyne.Position
 	Hidden   bool
 }
 
 // Get the current size of this widget.
 func (w *baseWidget) Size() fyne.Size {
+	//w.RLock()
+	//defer w.RUnlock()
 	return w.size
 }
 
 // Set a new size for a widget.
 // Note this should not be used if the widget is being managed by a Layout within a Container.
 func (w *baseWidget) resize(size fyne.Size, parent fyne.Widget) {
+	w.Lock()
```

Comment:
I didn't realize you could just inherit properties like this, but makes sense since everything is a struct.  Cool!